lib/honeycomb/propagation/aws.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require "base64"
+require "json"
+require "uri"
+
+module Honeycomb
+  # Parsing and propagation for AWS trace headers
+  module AWSPropagation
+    # Parse trace headers
+    module UnmarshalTraceContext
```

Comment:
This is what we're calling similar things in other beelines so I threw the `Unmarshal/Marshal` here for parity, but I don't think this is absolutely necessary, as it hasn't been a term in this beeline before. Totally up for discussion

Reply:
beeline-java uses `Encode` & `Decode` as part of the propagation codec design. I agree it's not totally necessary for it match others, but if we have the opportunity to synchronise, all the better for consistency 👍